src/app/pages/partners/results/results.tsx

```diff
@@ -0,0 +1,300 @@
+    if (!types.value) {
+        return candidates;
+    }
+    return candidates.filter((entry) => {
+        return (
+            types.value.localeCompare(entry.type, 'en', {
+                sensitivity: 'base'
+            }) === 0
+        );
+    });
+}
+
+type AdvancedType = ReturnType<typeof useSearchContext>['advanced'];
+
+// Custom advanced filter handling
+function filterBy(
+    values: string[],
+    candidates: PartnerEntry[],
+    candidateField: keyof PartnerEntry,
+    advanced: AdvancedType
+) {
+    const features = advanced.value.filter((f) => values.includes(f));
+
+    if (!features.length) {
+        return candidates;
+    }
+    return candidates.filter((entry) => {
+        const v = entry[candidateField] || '';
+
+        // Includes because for costs, the value is a semicolon-separated list
+        return features.some((f) => (v as unknown[]).includes(f));
+    });
+}
+
+// eslint-disable-next-line complexity
+function useFilteredEntries(entries: PartnerEntry[]) {
+    const {books, types, advanced, sort, resultCount} = useSearchContext();
+    const finalResult = React.useMemo(() => {
+        let result = shuffle(entries);
+
+        result = filterByBooks(result, books);
+        result = filterByType(result, types);
+
+        if (advanced.value.length > 0) {
+            result = result.filter((entry) => {
+                return advanced.value
+                    .filter((feature) => !costOptionValues.includes(feature))
+                    .filter((feature) => !equityOptionValues.includes(feature))
+                    .every((requiredFeature) =>
+                        entry.advancedFeatures.includes(requiredFeature)
+                    );
+            });
+            result = filterBy(costOptionValues, result, 'cost', advanced);
+            result = filterBy(
+                equityOptionValues,
+                result,
+                'equityRating',
+                advanced
+            );
+        }
+
+        return result;
+    }, [entries, advanced, books, types]);
+
+    resultCount.setValue(finalResult.length);
+
+    return React.useMemo(() => {
+        if (Math.abs(sort.value) === 0) {
+            return finalResult;
+        }
+
+        const getFieldsDict = {
+            '1': [(r: PartnerEntry) => r.title.toLowerCase()],
+            '2': [
+                (r: PartnerEntry) => Math.abs(r.rating),
+                (r: PartnerEntry) => r.ratingCount
+            ]
+        };
+        const sortDir = sort.value < 0 ? 'desc' : 'asc';
+
+        return orderBy(
+            finalResult,
+            getFieldsDict[Math.abs(sort.value) as unknown as '1' | '2'],
+            [sortDir, sortDir]
+        );
+    }, [finalResult, sort.value]);
+}
+
+function advancedFilterKeys(partnerEntry: PartnerData) {
+    return (Reflect.ownKeys(partnerEntry) as Array<keyof PartnerData>).filter(
+        (k) => ([false, true] as unknown[]).includes(partnerEntry[k])
+    );
+}
```

Comment:
There's no advantage here over Object.keys. I think Reflect is generally preferred for reflection. It will include (non-enumerable) symbol keys, while Object.keys can include inherited keys. Neither applies here.